### ISSUE TITLE
`hasOwnProperty` is not detected with `no-undef`

### ISSUE THREAD

**Issue by BridgeAR** [eslint/eslint#10757]
**Tell us about your environment**

* **ESLint Version:** 5.3.0
* **Node Version:** 11.0.0-pre
* **npm Version:** 6.2.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

```js
no-undef: error
```

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

[Demo](https://eslint.org/demo/#eyJ0ZXh0IjoiY29uc3QgZm9vID0gKCkgPT4ge1xyXHIgICAgaGFzT3duUHJvcGVydHkoKTtcclxyfTtcclxyZm9vKCk7XHIiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjgsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsiYWNjZXNzb3ItcGFpcnMiOjIsImFycmF5LWJyYWNrZXQtbmV3bGluZSI6MiwiYXJyYXktYnJhY2tldC1zcGFjaW5nIjoyLCJhcnJheS1jYWxsYmFjay1yZXR1cm4iOjIsImFycmF5LWVsZW1lbnQtbmV3bGluZSI6MiwiYXJyb3ctYm9keS1zdHlsZSI6MiwiYXJyb3ctcGFyZW5zIjoyLCJhcnJvdy1zcGFjaW5nIjoyLCJibG9jay1zY29wZWQtdmFyIjoyLCJibG9jay1zcGFjaW5nIjoyLCJicmFjZS1zdHlsZSI6MiwiY2FsbGJhY2stcmV0dXJuIjoyLCJjYW1lbGNhc2UiOjIsImNhcGl0YWxpemVkLWNvbW1lbnRzIjoyLCJjbGFzcy1tZXRob2RzLXVzZS10aGlzIjoyLCJjb21tYS1kYW5nbGUiOjIsImNvbW1hLXNwYWNpbmciOjIsImNvbW1hLXN0eWxlIjoyLCJjb21wbGV4aXR5IjoyLCJjb21wdXRlZC1wcm9wZXJ0eS1zcGFjaW5nIjoyLCJjb25zaXN0ZW50LXJldHVybiI6MiwiY29uc2lzdGVudC10aGlzIjoyLCJjb25zdHJ1Y3Rvci1zdXBlciI6MiwiY3VybHkiOjIsImRlZmF1bHQtY2FzZSI6MiwiZG90LWxvY2F0aW9uIjoyLCJkb3Qtbm90YXRpb24iOjIsImVxZXFlcSI6MiwiZm9yLWRpcmVjdGlvbiI6MiwiZnVuYy1jYWxsLXNwYWNpbmciOjIsImZ1bmMtbmFtZS1tYXRjaGluZyI6MiwiZnVuYy1uYW1lcyI6MiwiZnVuYy1zdHlsZSI6MiwiZnVuY3Rpb24tcGFyZW4tbmV3bGluZSI6MiwiZ2VuZXJhdG9yLXN0YXItc3BhY2luZyI6MiwiZ2V0dGVyLXJldHVybiI6MiwiZ2xvYmFsLXJlcXVpcmUiOjIsImd1YXJkLWZvci1pbiI6MiwiaGFuZGxlLWNhbGxiYWNrLWVyciI6MiwiaWQtYmxhY2tsaXN0IjoyLCJpZC1sZW5ndGgiOjIsImlkLW1hdGNoIjoyLCJpbXBsaWNpdC1hcnJvdy1saW5lYnJlYWsiOjIsImluZGVudC1sZWdhY3kiOjIsImluZGVudCI6MiwiaW5pdC1kZWNsYXJhdGlvbnMiOjIsImpzeC1xdW90ZXMiOjIsImtleS1zcGFjaW5nIjoyLCJrZXl3b3JkLXNwYWNpbmciOjIsImxpbmUtY29tbWVudC1wb3NpdGlvbiI6MiwibGluZXMtYXJvdW5kLWNvbW1lbnQiOjIsImxpbmVzLWFyb3VuZC1kaXJlY3RpdmUiOjIsImxpbmVzLWJldHdlZW4tY2xhc3MtbWVtYmVycyI6MiwibWF4LWNsYXNzZXMtcGVyLWZpbGUiOjIsIm1heC1kZXB0aCI6MiwibWF4LWxlbiI6MiwibWF4LWxpbmVzLXBlci1mdW5jdGlvbiI6MiwibWF4LWxpbmVzIjoyLCJtYXgtbmVzdGVkLWNhbGxiYWNrcyI6MiwibWF4LXBhcmFtcyI6MiwibWF4LXN0YXRlbWVudHMtcGVyLWxpbmUiOjIsIm1heC1zdGF0ZW1lbnRzIjoyLCJtdWx0aWxpbmUtY29tbWVudC1zdHlsZSI6MiwibXVsdGlsaW5lLXRlcm5hcnkiOjIsIm5ldy1jYXAiOjIsIm5ldy1wYXJlbnMiOjIsIm5ld2xpbmUtYWZ0ZXItdmFyIjoyLCJuZXdsaW5lLWJlZm9yZS1yZXR1cm4iOjIsIm5ld2xpbmUtcGVyLWNoYWluZWQtY2FsbCI6Miwibm8tYWxlcnQiOjIsIm5vLWFycmF5LWNvbnN0cnVjdG9yIjoyLCJuby1hc3luYy1wcm9taXNlLWV4ZWN1dG9yIjoyLCJuby1hd2FpdC1pbi1sb29wIjoyLCJuby1iaXR3aXNlIjoyLCJuby1idWZmZXItY29uc3RydWN0b3IiOjIsIm5vLWNhbGxlciI6Miwibm8tY2FzZS1kZWNsYXJhdGlvbnMiOjIsIm5vLWNhdGNoLXNoYWRvdyI6Miwibm8tY2xhc3MtYXNzaWduIjoyLCJuby1jb21wYXJlLW5lZy16ZXJvIjoyLCJuby1jb25kLWFzc2lnbiI6Miwibm8tY29uZnVzaW5nLWFycm93IjoyLCJuby1jb25zb2xlIjoyLCJuby1jb25zdC1hc3NpZ24iOjIsIm5vLWNvbnN0YW50LWNvbmRpdGlvbiI6Miwibm8tY29udGludWUiOjIsIm5vLWNvbnRyb2wtcmVnZXgiOjIsIm5vLWRlYnVnZ2VyIjoyLCJuby1kZWxldGUtdmFyIjoyLCJuby1kaXYtcmVnZXgiOjIsIm5vLWR1cGUtYXJncyI6Miwibm8tZHVwZS1jbGFzcy1tZW1iZXJzIjoyLCJuby1kdXBlLWtleXMiOjIsIm5vLWR1cGxpY2F0ZS1jYXNlIjoyLCJuby1kdXBsaWNhdGUtaW1wb3J0cyI6Miwibm8tZWxzZS1yZXR1cm4iOjIsIm5vLWVtcHR5LWNoYXJhY3Rlci1jbGFzcyI6Miwibm8tZW1wdHktZnVuY3Rpb24iOjIsIm5vLWVtcHR5LXBhdHRlcm4iOjIsIm5vLWVtcHR5IjoyLCJuby1lcS1udWxsIjoyLCJuby1ldmFsIjoyLCJuby1leC1hc3NpZ24iOjIsIm5vLWV4dGVuZC1uYXRpdmUiOjIsIm5vLWV4dHJhLWJpbmQiOjIsIm5vLWV4dHJhLWJvb2xlYW4tY2FzdCI6Miwibm8tZXh0cmEtbGFiZWwiOjIsIm5vLWV4dHJhLXBhcmVucyI6Miwibm8tZXh0cmEtc2VtaSI6Miwibm8tZmFsbHRocm91Z2giOjIsIm5vLWZsb2F0aW5nLWRlY2ltYWwiOjIsIm5vLWZ1bmMtYXNzaWduIjoyLCJuby1nbG9iYWwtYXNzaWduIjoyLCJuby1pbXBsaWNpdC1jb2VyY2lvbiI6Miwibm8taW1wbGljaXQtZ2xvYmFscyI6Miwibm8taW1wbGllZC1ldmFsIjoyLCJuby1pbmxpbmUtY29tbWVudHMiOjIsIm5vLWlubmVyLWRlY2xhcmF0aW9ucyI6Miwibm8taW52YWxpZC1yZWdleHAiOjIsIm5vLWludmFsaWQtdGhpcyI6Miwibm8taXJyZWd1bGFyLXdoaXRlc3BhY2UiOjIsIm5vLWl0ZXJhdG9yIjoyLCJuby1sYWJlbC12YXIiOjIsIm5vLWxhYmVscyI6Miwibm8tbG9uZS1ibG9ja3MiOjIsIm5vLWxvbmVseS1pZiI6Miwibm8tbG9vcC1mdW5jIjoyLCJuby1tYWdpYy1udW1iZXJzIjoyLCJuby1taXNsZWFkaW5nLWNoYXJhY3Rlci1jbGFzcyI6Miwibm8tbWl4ZWQtb3BlcmF0b3JzIjoyLCJuby1taXhlZC1yZXF1aXJlcyI6Miwibm8tbWl4ZWQtc3BhY2VzLWFuZC10YWJzIjoyLCJuby1tdWx0aS1hc3NpZ24iOjIsIm5vLW11bHRpLXNwYWNlcyI6Miwibm8tbXVsdGktc3RyIjoyLCJuby1tdWx0aXBsZS1lbXB0eS1saW5lcyI6Miwibm8tbmF0aXZlLXJlYXNzaWduIjoyLCJuby1uZWdhdGVkLWNvbmRpdGlvbiI6Miwibm8tbmVnYXRlZC1pbi1saHMiOjIsIm5vLW5lc3RlZC10ZXJuYXJ5IjoyLCJuby1uZXctZnVuYyI6Miwibm8tbmV3LW9iamVjdCI6Miwibm8tbmV3LXJlcXVpcmUiOjIsIm5vLW5ldy1zeW1ib2wiOjIsIm5vLW5ldy13cmFwcGVycyI6Miwibm8tbmV3IjoyLCJuby1vYmotY2FsbHMiOjIsIm5vLW9jdGFsLWVzY2FwZSI6Miwibm8tb2N0YWwiOjIsIm5vLXBhcmFtLXJlYXNzaWduIjoyLCJuby1wYXRoLWNvbmNhdCI6Miwibm8tcGx1c3BsdXMiOjIsIm5vLXByb2Nlc3MtZW52IjoyLCJuby1wcm9jZXNzLWV4aXQiOjIsIm5vLXByb3RvIjoyLCJuby1wcm90b3R5cGUtYnVpbHRpbnMiOjIsIm5vLXJlZGVjbGFyZSI6Miwibm8tcmVnZXgtc3BhY2VzIjoyLCJuby1yZXN0cmljdGVkLWdsb2JhbHMiOjIsIm5vLXJlc3RyaWN0ZWQtaW1wb3J0cyI6Miwibm8tcmVzdHJpY3RlZC1tb2R1bGVzIjoyLCJuby1yZXN0cmljdGVkLXByb3BlcnRpZXMiOjIsIm5vLXJlc3RyaWN0ZWQtc3ludGF4IjoyLCJuby1yZXR1cm4tYXNzaWduIjoyLCJuby1yZXR1cm4tYXdhaXQiOjIsIm5vLXNjcmlwdC11cmwiOjIsIm5vLXNlbGYtYXNzaWduIjoyLCJuby1zZWxmLWNvbXBhcmUiOjIsIm5vLXNlcXVlbmNlcyI6Miwibm8tc2hhZG93LXJlc3RyaWN0ZWQtbmFtZXMiOjIsIm5vLXNoYWRvdyI6Miwibm8tc3BhY2VkLWZ1bmMiOjIsIm5vLXNwYXJzZS1hcnJheXMiOjIsIm5vLXN5bmMiOjIsIm5vLXRhYnMiOjIsIm5vLXRlbXBsYXRlLWN1cmx5LWluLXN0cmluZyI6Miwibm8tdGVybmFyeSI6Miwibm8tdGhpcy1iZWZvcmUtc3VwZXIiOjIsIm5vLXRocm93LWxpdGVyYWwiOjIsIm5vLXRyYWlsaW5nLXNwYWNlcyI6Miwibm8tdW5kZWYtaW5pdCI6Miwibm8tdW5kZWYiOjIsIm5vLXVuZGVmaW5lZCI6Miwibm8tdW5kZXJzY29yZS1kYW5nbGUiOjIsIm5vLXVuZXhwZWN0ZWQtbXVsdGlsaW5lIjoyLCJuby11bm1vZGlmaWVkLWxvb3AtY29uZGl0aW9uIjoyLCJuby11bm5lZWRlZC10ZXJuYXJ5IjoyLCJuby11bnJlYWNoYWJsZSI6Miwibm8tdW5zYWZlLWZpbmFsbHkiOjIsIm5vLXVuc2FmZS1uZWdhdGlvbiI6Miwibm8tdW51c2VkLWV4cHJlc3Npb25zIjoyLCJuby11bnVzZWQtbGFiZWxzIjoyLCJuby11bnVzZWQtdmFycyI6Miwibm8tdXNlLWJlZm9yZS1kZWZpbmUiOjIsIm5vLXVzZWxlc3MtY2FsbCI6Miwibm8tdXNlbGVzcy1jb21wdXRlZC1rZXkiOjIsIm5vLXVzZWxlc3MtY29uY2F0IjoyLCJuby11c2VsZXNzLWNvbnN0cnVjdG9yIjoyLCJuby11c2VsZXNzLWVzY2FwZSI6Miwibm8tdXNlbGVzcy1yZW5hbWUiOjIsIm5vLXVzZWxlc3MtcmV0dXJuIjoyLCJuby12YXIiOjIsIm5vLXZvaWQiOjIsIm5vLXdhcm5pbmctY29tbWVudHMiOjIsIm5vLXdoaXRlc3BhY2UtYmVmb3JlLXByb3BlcnR5IjoyLCJuby13aXRoIjoyLCJub25ibG9jay1zdGF0ZW1lbnQtYm9keS1wb3NpdGlvbiI6Miwib2JqZWN0LWN1cmx5LW5ld2xpbmUiOjIsIm9iamVjdC1jdXJseS1zcGFjaW5nIjoyLCJvYmplY3QtcHJvcGVydHktbmV3bGluZSI6Miwib2JqZWN0LXNob3J0aGFuZCI6Miwib25lLXZhci1kZWNsYXJhdGlvbi1wZXItbGluZSI6Miwib25lLXZhciI6Miwib3BlcmF0b3ItYXNzaWdubWVudCI6Miwib3BlcmF0b3ItbGluZWJyZWFrIjoyLCJwYWRkZWQtYmxvY2tzIjoyLCJwYWRkaW5nLWxpbmUtYmV0d2Vlbi1zdGF0ZW1lbnRzIjoyLCJwcmVmZXItYXJyb3ctY2FsbGJhY2siOjIsInByZWZlci1jb25zdCI6MiwicHJlZmVyLWRlc3RydWN0dXJpbmciOjIsInByZWZlci1udW1lcmljLWxpdGVyYWxzIjoyLCJwcmVmZXItb2JqZWN0LXNwcmVhZCI6MiwicHJlZmVyLXByb21pc2UtcmVqZWN0LWVycm9ycyI6MiwicHJlZmVyLXJlZmxlY3QiOjIsInByZWZlci1yZXN0LXBhcmFtcyI6MiwicHJlZmVyLXNwcmVhZCI6MiwicHJlZmVyLXRlbXBsYXRlIjoyLCJxdW90ZS1wcm9wcyI6MiwicXVvdGVzIjoyLCJyYWRpeCI6MiwicmVxdWlyZS1hdG9taWMtdXBkYXRlcyI6MiwicmVxdWlyZS1hd2FpdCI6MiwicmVxdWlyZS11bmljb2RlLXJlZ2V4cCI6MiwicmVxdWlyZS15aWVsZCI6MiwicmVzdC1zcHJlYWQtc3BhY2luZyI6Miwic2VtaS1zcGFjaW5nIjoyLCJzZW1pLXN0eWxlIjoyLCJzZW1pIjoyLCJzb3J0LWltcG9ydHMiOjIsInNvcnQta2V5cyI6Miwic29ydC12YXJzIjoyLCJzcGFjZS1iZWZvcmUtYmxvY2tzIjoyLCJzcGFjZS1iZWZvcmUtZnVuY3Rpb24tcGFyZW4iOjIsInNwYWNlLWluLXBhcmVucyI6Miwic3BhY2UtaW5maXgtb3BzIjoyLCJzcGFjZS11bmFyeS1vcHMiOjIsInNwYWNlZC1jb21tZW50IjoyLCJzdHJpY3QiOjIsInN3aXRjaC1jb2xvbi1zcGFjaW5nIjoyLCJzeW1ib2wtZGVzY3JpcHRpb24iOjIsInRlbXBsYXRlLWN1cmx5LXNwYWNpbmciOjIsInRlbXBsYXRlLXRhZy1zcGFjaW5nIjoyLCJ1bmljb2RlLWJvbSI6MiwidXNlLWlzbmFuIjoyLCJ2YWxpZC1qc2RvYyI6MiwidmFsaWQtdHlwZW9mIjoyLCJ2YXJzLW9uLXRvcCI6Miwid3JhcC1paWZlIjoyLCJ3cmFwLXJlZ2V4IjoyLCJ5aWVsZC1zdGFyLXNwYWNpbmciOjIsInlvZGEiOjJ9LCJlbnYiOnt9fX0=)

**What did you expect to happen?**

An error should be reported.

**What actually happened? Please include the actual, raw output from ESLint.**

Nothing is reported. It is specific to this function name.

**Comment by Aladdin-ADD** [comment]
seems a bug to me. Thanks for the report!

**Comment by not-an-aardvark** [comment]
I think this is working as intended because the global object typically has `Object.prototype` on its prototype chain, so referencing properties `hasOwnProperty` will read that value. I've seen `hasOwnProperty.call(foo, bar)` used as an alternative to `Object.prototype.hasOwnProperty.call(foo, bar)`.

**Comment by BridgeAR** [comment]
@not-an-aardvark in browsers it does seem to be attached to the window. But in Node.js that is not true and it will very likely not change. So IMO at least in combination with `env: { node: true }` it should be considered an error.

**Comment by not-an-aardvark** [comment]
It also works in Node:

```
$ node -e 'console.log(hasOwnProperty)'
[Function: hasOwnProperty]
```

This works in any environment where the global object inherits from `Object.prototype`. The only cases where it wouldn't work would be when the global object is created with something like `Object.create(null)`.

**Comment by BridgeAR** [comment]
Oh, you are right. Thanks for pointing this out! I just ran into an issue with it because I used `uncurryThis` to prevent tampering of the globals... I guess this is a special case without an ideal solution.
